pkg/kubelet/types/pod_status.go

```diff
@@ -42,5 +42,10 @@ func PodConditionByKubelet(conditionType v1.PodConditionType) bool {
 			return true
 		}
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
+		if conditionType == v1.AlphaNoCompatGuaranteeDisruptionTarget {
```

Comment:
maybe we shouldn't check the gate here? so that if the feature is disabled, the kubelet can remove the condition in the next iteration.

Reply:
I think we need to be very cautious about adding code unprotected by the feature gate flag, cause in case of problems there would be no way to disable it. I'm trying to understand what is the problematic scenario. Is it that someone runs kubelet with the gate enabled, it results in adding the pod condition, the he/she restarts kubelet with the gate disabled? I think it is a corner case but leaving the condition added seems reasonable.

Reply:
> I think we need to be very cautious about adding code unprotected by the feature gate flag

That's the general idea, yes. But sometimes you have to leave cleanup code unprotected. That increases the risk of a feature, but it needs to be done in some cases.

> Is it that someone runs kubelet with the gate enabled, it results in adding the pod condition, the he/she restarts kubelet with the gate disabled? I think it is a corner case but leaving the condition added seems reasonable.

yeah, you are right, it seems reasonable for a condition like this.